TypeError in test_relative_to() with Python 3.15

A test build of the Fedora plumbum package using Python 3.15 caused the following error. See also https://bugzilla.redhat.com/show_bug.cgi?id=2424319. Is this related to https://github.com/tomerfiliba/plumbum/issues/673?

```
self = <test_local.TestLocalPath object at 0x7faf6b9265d0>

    def test_relative_to(self):
        p = local.path("/var/log/messages")
        assert p.relative_to("/var/log/messages") == RelativePath([])
        assert p.relative_to("/var/") == RelativePath(["log", "messages"])
        assert p.relative_to("/") == RelativePath(["var", "log", "messages"])
        assert p.relative_to("/var/tmp") == RelativePath(["..", "log", "messages"])
        assert p.relative_to("/opt") == RelativePath(["..", "var", "log", "messages"])
        assert p.relative_to("/opt/lib") == RelativePath(
            ["..", "..", "var", "log", "messages"]
        )
        for src in [
            local.path("/var/log/messages"),
            local.path("/var"),
            local.path("/opt/lib"),
        ]:
            delta = p.relative_to(src)
>           assert src + delta == p
                   ^^^^^^^^^^^
E           TypeError: can only concatenate str (not "RelativePath") to str

delta      = RelativePath([])
p          = <LocalPath /var/log/messages>
self       = <test_local.TestLocalPath object at 0x7faf6b9265d0>
src        = <LocalPath /var/log/messages>

/builddir/build/BUILD/python-plumbum-1.10.0-build/plumbum-1.10.0/tests/test_local.py:136: TypeError
```

--- a/plumbum/path/base.py
+++ b/plumbum/path/base.py
@@ -50,14 +50,6 @@ class Path(str, ABC):
         """Joins two paths"""
         return self.join(other)
 
-    def __add__(self, other: str | Self) -> Self:
-        """Attaches text to path"""
-        return self.__class__(str(self) + str(other))
-
-    def __radd__(self, other: str) -> Self:
-        """Attaches text to path"""
-        return self.__class__(other + str(self))
-
     @typing.overload
     def __getitem__(self, key: str | Path) -> Self: ...
 

--- a/tests/test_local.py
+++ b/tests/test_local.py
@@ -133,7 +133,7 @@ class TestLocalPath:
             local.path("/opt/lib"),
         ]:
             delta = p.relative_to(src)
-            assert src + delta == p
+            assert src / delta == p
 
     def test_read_write(self):
         with local.tempdir() as dir:

